test/integrationtests/voight_kampff/features/steps/utterance_responses.py

```diff
@@ -92,17 +119,23 @@ def dialog_from_sentence(sentence, skill_path, lang):
 def _match_dialog_patterns(dialogs, sentence):
     """Match sentence against a list of dialog patterns.
 
-    Returns index of found match.
+    dialogs (list of str): dialog file entries to match against
+    sentence (str): string to match.
+
+    Returns:
+        (tup) index of found match, debug text
     """
     # Allow custom fields to be anything
-    dialogs = [re.sub(r'{.*?\}', r'.*', dia) for dia in dialogs]
-    # Remove left over '}'
-    dialogs = [re.sub(r'\}', r'', dia) for dia in dialogs]
-    dialogs = [re.sub(r' .* ', r' .*', dia) for dia in dialogs]
-    # Merge consequtive .*'s into a single .*
-    dialogs = [re.sub(r'\.\*( \.\*)+', r'.*', dia) for dia in dialogs]
-    # Remove double whitespaces
-    dialogs = ['^' + ' '.join(dia.split()) for dia in dialogs]
+    # i.e {field} gets turned into ".*"
+    regexes = []
+    for dialog in dialogs:
+        data = {element[1]: '.*'
+                for element in Formatter().parse(dialog)}
+        regexes.append(dialog.format(**data))
+
+    # Remove double whitespaces and ensure that it matches from
+    # the beginning of the line.
+    regexes = ['^' + ' '.join(reg.split()) for reg in regexes]
```

Comment:
unused? I'm trying to debug through why this works but doesn't output anything... strange.

Reply:
I think the start of line anchor is usable but the split may not be. (I thought the expand_options could leave extra ' ' but it seems it doesn't). I haven't have time to do many more tests today so it'll be a weekend project.

I'll remove the space cleanup, thanks for reviewing and finding improvements!

Reply:
Right I just meant the regexes variable isn't consumed later in the function.

Thanks for working on this issue!

Reply:
*face palm* doh, right you are. That would have screwed up the pattern matching of data entries completely. thank you!

I've amended the offending commit. but will likely do the actual testing tomorrow.

Reply:
OK! I pulled the amended commit and at least for my limited set of dialog files, it's working correctly.

Reply:
The tests I did also looks good.